crates/sui-json-rpc/src/lib.rs

```diff
@@ -153,6 +156,23 @@ impl JsonRpcServerBuilder {
         let routing_layer = RoutingLayer::new(routing, disable_routing);
 
         let middleware = tower::ServiceBuilder::new()
+            .layer(
+                TraceLayer::new_for_http()
+                    .make_span_with(|request: &Request<Body>| {
+                        let request_id = request
+                            .headers()
```

Comment:
does this fail gracefully if the header isn't present?  just checking

Reply:
yep it does!